### DISCUSSION TITLE
ARITHMETIC CALCULATIONS

### DISCUSSION THREAD

**Discussion by Jijiri89** [filamentphp/filament#7181] [General]
Hello
Please can anyone kindly help me how to do arithmetic calculations in filament?
For example, I have 3 columns in my laravel database (they are, quantity, un-price and price columns) I want a way to multiple quantities by un-price and store the result in price column.
How can I do that?

**Comment by leandrocfe** [comment]
maybe a virtual/stored column?
https://pineco.de/using-computed-columns-in-laravel/

**Reply by Jijiri89** [reply]
Thanks so much. It worked

**Comment by emcarson** [comment]
Maybe this Filament Trick will help you:

```php
Tables\Columns\TextColumn::make('myCalculatedColumn')
    ->getStateUsing(function(Model $record) {
        // return whatever you need to show
        return $record->field1 + $record->field2;
    })
```

Source: https://filamentphp.com/tricks/calculated-value-in-column
Author: Juan Benítez

PS: If you want to be able to sort by the calculated column I recommend @leandrocfe's solution.
Or create a real column and update it automatically using a Model saving event.
